### PR TITLE
Remove all NIP authors

### PR DESCRIPTION
**Pull request by fiatjaf**
I don't know if this is a good idea, but it might be. Let me know.

I think having people's names in NIPs cause others to be afraid of proposing changes or additions to them, and also NIPs are all so heavily modified by everybody, either directly or indirectly, that these names end up not really reflecting reality at all.

### CODE REVIEW

**Comment by staab** [comment]
It's always seemed unnecessary to me. Everything is public domain anyway.

**Review by alexgleason — APPROVED** [review]
This is a good change. `git blame` solves any practical questions better than a list of authors. Removing it simplifies things.

**Comment by vitorpamplona** [comment]
I would remove status as well. Nothing is "final" and nothing is "mandatory".

**Comment by staab** [comment]
> I would remove status as well. Nothing is "final" and nothing is "mandatory".

Yeah, status is a joke, I exclusively copy/paste headers when drafting new nips and never update it. 😂

**Comment by mikedilger** [comment]
I ain't got no problems

**Comment by erskingardner** [comment]
ACK – great idea. we're all shadowy super coders anyway... 🥷

**Comment by gzuuus** [comment]
ACK

**Comment by ursuscamp** [comment]
ACK.

NIPs are more like living documents than the BIP-style record of history. Authors and even status make little sense.

**Comment by melvincarvalho** [comment]
It's fairly standard to have authors on specification like proposals, so that you know who to talk to, at a glance.  I see this got railroaded through anyway.

Main reason for -1 is that a lot of authors put a lot of effort into the work, and you dont have consent to remove them.

**Comment by vitorpamplona** [comment]
> It's fairly standard to have authors on specifications like proposals, so that you know who to talk to, at a glance. 

It's a double-edged sword. Having authors increases the overly defensive behavior of the writer, even in the face of new evidence to justify a change. Authors tend to not update their NIPs with new use cases to keep lazily their own implementations compliant with their NIPs, even though no one uses them in the way they described anymore. 

It also creates this weird behavior of "I am a god" because everyone is implementing "my" NIP. 

The NIP text is just a reflection of what's out there in code. It's no one's property. Even if you start new event kinds, the NIP must reflect how everyone else is using it. It doesn't matter if you are an author and the first implementer anymore.

**Comment by melvincarvalho** [comment]
Three process issues with this

1.  It's bike-shedding (obviously) - so many more pressing things in the project to be done, such as more transparency, not less
2.  There was no author consent
3.  12 hours is not enough time

Regarding authors, I can see both sides of argument, but the pro author was not really represented.  I'd like to still be an author on NIP-98 although Kieran did most of the work.  Reasons why having an author, which is best practice in standards, might be good:

Having specific authors for specifications, as opposed to relying on tools like "git blame," offers several significant advantages:

1. **Clear Responsibility and Expertise**: Assigning authors to specifications clearly identifies who is responsible for the content and whom to consult for expert insights. This clarity is particularly crucial when dealing with complex or technical subjects.

2. **Accountability and Ownership**: When individuals are named as authors, they are more likely to take ownership of the quality and accuracy of the information. This sense of responsibility can lead to higher-quality specifications.

3. **Historical Context and Intent**: Authors can provide context and rationale behind certain decisions, which might not be apparent from the document alone. This background is especially valuable in understanding the intent behind specific choices or approaches.

4. **Long-Term Maintenance and Updates**: Authors can be consulted for future updates or revisions. Their involvement ensures continuity and coherence in the evolution of the specification.

5. **Avoiding Misinterpretation**: Tools like "git blame" can sometimes lead to misinterpretations, as they only show who last modified a line, not the context or reasoning behind the change. Direct authorship provides a more accurate and comprehensive understanding.

6. **Collaborative Work Recognition**: Specifications often result from collaborative efforts. Acknowledging authors respects and recognizes the contributions of those involved in the development process.

7. **Professional Development and Recognition**: Being named as an author can be significant for the professional development and recognition of the contributors, encouraging further contributions and involvement in the field.

8. **Quality Control**: Authorship often involves a review process, which can significantly enhance the quality and reliability of the specifications.

In summary, having named authors for specifications ensures clarity, responsibility, and quality in the creation and maintenance of these important documents. It facilitates better communication, historical understanding, and professional recognition, leading to more robust and reliable specifications.

**Comment by AsaiToshiya** [comment]
> > Long-Term Maintenance and Updates
> 
> Absolutely false. Again, it's was we want to believe. But in practice, very few authors come back to refine their own texts.

This is correct. Actually, the author of NIP-36 did not reply to https://github.com/nostr-protocol/nips/issues/769#issuecomment-1741215224.

**Comment by staab** [comment]
@melvincarvalho I know you've been around these parts longer than I have, but it doesn't seem like you understand what nostr the protocol is, or what the NIPs repo is. The protocol is "what is interoperable based on real implementations and usage", which is hard (impossible?) to quantify. The NIPs repo, like a constitution, is descriptive of the state of the world as seen by the consensus emerging from the repo's contributors, not prescriptive of the state of the world. That's why it's public domain, to detach ego and ownership from the protocol. This isn't fiatjaf's work, or anyone else's.

I'm not 100% sure this model really makes sense, or is actually possible to execute. But that's the way it's been framed for as long as I've been around. Your points only apply to protocols that are someone's intellectual property, or are designed top-down by a committee. Therefore, your implicit claim is that nostr is top-down and designed by committee. Maybe you're right! But if it is an un-protocol, as it claims to be, most of your points (here, and about transparent authorship) don't actually apply.

In theory, this protocol has no individual authors, only emergent behaviors loosely documented for posterity. In that sense, your comparison of people with merge access to a priesthood is apt, since priests are the stewards of sacred knowledge. In this case though the knowledge is not revealed from on high, but supplied by everyone involved in the protocol (even non-developers contribute in important ways). The NIPs repo is not authoritative, because it is unavoidably inaccurate and incomplete. However, it is the best representation we have, based on the experience and wisdom of everyone who contributes to it. This set of people has to be guarded, because not everyone has experience and wisdom. And the only way I know to select these people is through authority (maybe nostrocket can fix that).

The important thing to understand for making this system work (if indeed it can work and isn't an ancap fantasy) is that anyone can open an issue or PR, write code, attack the network, or complain loudly. The protocol relies on people (not just incumbents) to do that! Also, anyone can fork this repo, add their own contributors, and market it better if they want to become the source of truth. Of course, forks cause a lot of damage (e.g. BCH and the Schism of 1054), so I appreciate that you're expressing your concerns here first.

That's my two cents, but [terms may apply](nostr:nevent1qqsf5qz80hxlgg4p8kxh98alx249k4skya5cenpe9y7g667fxgtg7jqpz4mhxue69uhhyetvv9ujuerpd46hxtnfduhsz9rhwden5te0wfjkcctev93xcefwdaexwtcpzdmhxue69uhk7enxvd5xz6tw9ec82c30qyw8wumn8ghj7cn4vd4k2apwvdhhyctrd3jjuum0vd5kzmp0qy88wumn8ghj7mn0wvhxcmmv9uq3xamnwvaz7tmsw4e8qmr9wpskwtn9wvhsz8rhwden5te0vd58y6tnw3cxjmrv9ehx7um5wgcjucm0d5hszxthwden5te0wfjkccte9eekummjwsh8xmmrd9skctcpzemhxue69uhhyetvv9ujumt0wd68ytnsw43z7qgjwaehxw309ac82unsd3jhqct89ejhxpenvm5)

**Comment by jb55** [comment]
I'm pretty proud of the zaps spec as a put a lot of thinking into it. sad to see history getting erased.

**Comment by hsjoberg** [comment]
NACK. It's useful information to know who has worked on different specifications and it can also be a thing of pride for people.

**Comment by mleku** [comment]
if this repo is zipped up for a release, the git log is removed.

if this repo is distributed without copying the entire folder recursively, just the contents of the folder, the git log is removed, unless the dotfiles in it are specified.

this is also a violation of the core principle of keeping things in plaintext and not requiring binary interpretation for data whenever it is practical, storage being a key time of this.

in the light of the documents therein containing justifications for why a plaintext format for the messaging protocol is present, and the plaintext version of attributions is now not, this is very concerning.

also it is concerning that this was merged without the explicit consent of everyone whose name was mentioned.

**Comment by melvincarvalho** [comment]
Main concern is that Zebedee can railroad through changes to 57 NIPs in 12 hours, in an _open_ protocol.  That is much too short, IMHO.

I've kept a copy of the author history here in case there's any interest:

https://github.com/nostr-nips/nips

Let me note that author doesnt always correlate to git blame, because many things are designed in person, or through conversations outside of github.  It's a shame that those people who contributed to the history of nostr get cancelled, without an appeal.  But it is what it is.

It would be nice to keep that repo up to date with the official NIPs repo, but in reality, it's unlikely to happen.

That said, if anyone here that wants to volunteer or has a neat script to keep things up to date, is welcome to have merge access.

Another use for the unofficial repo, aside from keeping author information, could be to have status=experimental.  We have no where to do that right now, so could be an option.

**Comment by melvincarvalho** [comment]
IMHO a good thing to prevent things like this happening again, would be for the repo to have a 2nd owner, other than zebedee, to provide a counter balance and sanity check.  One person, no matter how well meaning, is always going to make mistakes or have conflicts of interest.  Building a truly open protocol without a single point of failure would be a better way of doing things.

**Comment by gzuuus** [comment]
> Main concern is that Zebedee can railroad through changes to 57 NIPs in 12 hours, in an open protocol. That is much too short, IMHO.

Yes, it is a bit worrying, I would also have liked to have had more time to reach a greater consensus, as there are people who were bothered by it.
My ACK was because I specifically did not mind my name appearing in the repository, because I think that in an open protocol project it is important to separate the power or ability to decide from certain people, or with an attributed merit for being a contributor, I think that the development of the protocol needs committed people who can participate publicly, not privately, as the private consultation of contributors may favour. 
However, the way this event has unfolded makes it clear that we need a more refined and collaborative model of consensus, otherwise everything is literally hanging by a thread.
It may be worth having a "CONTRIBUTORS" file in the repository, to provide a common ground between different points of view, to allow transparency in development, and to acknowledge contributors without cluttering up the main documentation.

**Comment by mattn** [comment]
How about to put CODE_OWNER file on this repository?

example
```
*.md @fiatjaf
27.md @mattn
```

**Comment by weex** [comment]
> How about to put CODEOWNERS file on this repository?

It's better to revert this and dig into the conditions that made this merge possible.

**Comment by vitorpamplona** [comment]
> Nostr today cannot call itself an open protocol. Not even close. Bitcoin would never do this, fediverse would never do this, bluesky would never do this.

Repeat with me: The NIP repo is NOT "the protocol". 

Nostr is open because you can do whatever you want REGARDLESS of the NIP repo. The NIP repo is just a way for us to document what is happening out there and get to some common language about our stuff. That's it. It's nothing more than that. It's a nice to have. It's a town hall for discussions, not a regulatory body. 

> BIPs have authors. They are based on python PEPs that have authors. FEPs have authors. 

Sure, but they are not required. The US Code, for instance, doesn't have any authors. Because it doesn't make any sense to have authors in highly collaborative texts. Authors are present only on Bills, which are like a git commit for us. And those are the "sponsors", not really the writers.

**Comment by jb55** [comment]
On Fri, Nov 17, 2023 at 07:37:28AM -0800, Vitor Pamplona wrote:
>> BIPs have authors. They are based on python PEPs that have authors.
>> FEPs have authors.
>
>Sure, but they are not required. The US Code, for instance, doesn't
>have any authors. Because it doesn't make any sense to have authors in
>highly collaborative texts. Authors are present only on Bills, which
>are like a git commit for us. And those are the "sponsors", not really
>the writers.

Still pretty useful to know who to contact to ask questions regarding
the spec. Usually the people who worked on it understand the
implications of certain things, especially wrt. to the zaps spec which
has non-trivial validation requirements.

**Comment by staab** [comment]
> also it is concerning that this was merged without the explicit consent of everyone whose name was mentioned.

The NIPs repo is public domain. Authors contributed to it as public domain. This is implicit consent to it all being changed, or deleted, or replicated, in any way, without further consent. If explicit consent is the bar, we need to have everyone sign contracts waiving their right to their IP like Cognitect does with clojure.

> It may be worth having a "CONTRIBUTORS" file in the repository

I think this would be fine. Anyone who wants credit can have it (and Will, everyone knows you created zaps, and not because of the nips repo). We could also add "owners" to that list so we don't need to rely on github's mechanism which requires owners to mark themselves as public.

**Comment by vitorpamplona** [comment]
> Still pretty useful to know who to contact to ask questions regarding the spec. Usually, the people who worked on it understand the implications of certain things, especially wrt. to the zaps spec which has non-trivial validation requirements.

Authors can definitely help if they want. But I don't think it's a good idea to centralize inquiries on authors: 
1. It's centralizing in nature. The whole point of Nostr is to not do this. 
2. Few authors have the time to discuss/explain and are usually terrible educators. 
3. If users are reaching out to ask/propose changes, an issue/PR offers a clearer/more formal path to explain **exactly** what they intend to do. Then the entire community can see/reply.
4. If the author is still motivated to work on their NIP (most are not), they can keep following the discussion here. Otherwise, they are free to go.

**Comment by staab** [comment]
I'm going to go ahead and link my post from a related issue just so people can decide how seriously to take @melvincarvalho here. https://github.com/nostr-protocol/nips/issues/162#issuecomment-1815151208

I do appreciate that he has walked back his tone since then. The comparison with BIPs is helpful. I have no experience running an open source protocol, and I imagine neither do most of us. Ripping off bitcoin's approach seems like a step in the right direction.

**Comment by pablof7z** [comment]
There's a major difference with BIPs in that bitcoin requires consensus, so that's a very tough analogy to work with imo.

**Comment by pablof7z** [comment]
> > Still pretty useful to know who to contact to ask questions regarding the spec. Usually, the people who worked on it understand the implications of certain things, especially wrt. to the zaps spec which has non-trivial validation requirements.
> 
> Authors can definitely help if they want. But I don't think it's a good idea to centralize inquiries on authors:
> 
>     1. It's centralizing in nature. The whole point of Nostr is to not do this.
> 
>     2. Few authors have the time to discuss/explain and are usually terrible educators.
> 
>     3. If devs are reaching out to ask/propose changes, an issue/PR offers a clearer/more formal path to explain **exactly** what they intend to do. Then the entire community can see/reply.
> 
>     4. If the author is still motivated to work on their NIP (most are not), they can keep following the discussion here. Otherwise, they are free to go.

so much this 👆

🤝

**Comment by mleku** [comment]
> There's a major difference with BIPs in that bitcoin requires consensus, so that's a very tough analogy to work with imo.


> > > Still pretty useful to know who to contact to ask questions regarding the spec. Usually, the people who worked on it understand the implications of certain things, especially wrt. to the zaps spec which has non-trivial validation requirements.
> > 
> > 
> > Authors can definitely help if they want. But I don't think it's a good idea to centralize inquiries on authors:
> > 
> > 1. It's centralizing in nature. The whole point of Nostr is to not do this.
> > 2. Few authors have the time to discuss/explain and are usually terrible educators.
> > 3. If devs are reaching out to ask/propose changes, an issue/PR offers a clearer/more formal path to explain **exactly** what they intend to do. Then the entire community can see/reply.
> > 4. If the author is still motivated to work on their NIP (most are not), they can keep following the discussion here. Otherwise, they are free to go.
> 
> so much this 👆
> 
> 🤝

Without any common ground there will be forks.

Businesses need some basic common ground to be able to propose plans to their funders to make the stuff so it can be put out there and make a difference in the world. Without businesses adopting this technology you can forget about it having a future in 3 years time.

Only individuals act.

Collectives are just a cloak for people who don't want to be held responsible.
